engine/components-api/src/main/java/com/cloud/storage/StorageManager.java

```diff
@@ -163,6 +163,36 @@ public interface StorageManager extends StorageService {
             true,
             ConfigKey.Scope.Cluster,
             null);
+    ConfigKey<Boolean> VmwareCreateCloneFull = new ConfigKey<>(Boolean.class,
+            "vmware.create.full.clone",
+            "Storage",
+            "false",
+            "If set to true, creates VMs as full clones on ESX hypervisor",
+            true,
+            ConfigKey.Scope.StoragePool,
+            null);
+    ConfigKey<Boolean> VmwareAllowParallelExecution = new ConfigKey<>(Boolean.class,
+            "vmware.allow.parallel.command.execution",
+            "Advanced",
+            "false",
+            "allow commands to be executed in parallel in spite of 'vmware.create.full.clone' being set to true.",
+            true,
+            ConfigKey.Scope.Global,
+            null);
+
+    static boolean shouldExecuteInSequenceOnVmware() {
+                final Boolean fullClone = getFullCloneConfiguration();
+                final Boolean allowParallel = getAllowParallelExecutionConfiguration();
+                return fullClone && !allowParallel;
+            }
+
+    static Boolean getAllowParallelExecutionConfiguration() {
+        return VmwareAllowParallelExecution.value();
+    }
+
+    static Boolean getFullCloneConfiguration() {
```

Comment:
Since this configuration is storage-pool wide, shouldn't this method check the value a specific pool? Otherwise it is taking the global value but not the granular values

Reply:
@nvazquez what if there is a command with a destination and a source that have a different setting for create full clone? How do we decide? || && or something else?